### PR TITLE
OCPBUGS-17424: Handle cache.DeletedFinalStateUnknown correctly

### PR DESCRIPTION
**Pull request by ncdc**

### CODE REVIEW

**Comment by wking** [comment]
Run the suite that panicked, to ensure we excercise the adjusted ConfigMap delete handler (even if we probably miss the well-timed watch-interruption that triggered [this panic][1]):

/payload-job periodic-ci-openshift-release-master-ci-4.14-e2e-aws-sdn-serial

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.14-e2e-aws-sdn-serial/1687876857824808960

**Comment by ncdc** [comment]
```
Cluster operator insights SCAAvailable is False with Forbidden: Failed to pull SCA certs from https://api.openshift.com/api/accounts_mgmt/v1/certificates: OCM API https://api.openshift.com/api/accounts_mgmt/v1/certificates returned HTTP 403: {"code":"ACCT-MGMT-11","href":"/api/accounts_mgmt/v1/errors/11","id":"11","kind":"Error","operation_id":"d37a48c3-c5df-426c-a595-3a178d40e1ad","reason":"Account with ID 2DUeKzzTD9ngfsQ6YgkzdJn1jA4 denied access to perform create on Certificate with HTTP call POST /api/accounts_mgmt/v1/certificates"}
```

/retest

File: pkg/console/controllers/util/util.go [hunk 1]
```diff
@@ -8,21 +8,29 @@ import (
...
+
+	"github.com/openshift/console-operator/pkg/api"
 )
 
 // Return func which returns true if obj name is in names
 func IncludeNamesFilter(names ...string) factory.EventFilterFunc {
 	nameSet := sets.NewString(names...)
 	return func(obj interface{}) bool {
-		metaObj := obj.(metav1.Object)
+		if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); ok {
```
Comment (jhadvig) [inline pkg/console/controllers/util/util.go:27]: Could you please also add this logic to the `LabelFilter` function below.

**Review by jhadvig — COMMENTED** [review]
@ncdc thanks for this fix!
Added one comment. PTAL

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Comment by ncdc** [comment]
```
{  fail [github.com/openshift/origin/test/extended/apiserver/api_requests.go:449]: Expected
    <[]string | len:1, cap:1>: [
        "Operator \"kube-controller-manager-operator\" produces more watch requests than expected: watchrequestcount=335, upperbound=330, ratio=1.02",
    ]
to be empty
Ginkgo exit error 1: exit with code 1}
```

Unrelated failure

/retest

**Pull request merged**